### PR TITLE
Update API configuration

### PR DESCRIPTION
**Pull request by pantherale0**
This updates the user agent and associated API endpoints to resolve https://github.com/home-assistant/core/issues/105816?

### CODE REVIEW

**Comment by dcmeglio** [comment]
I tried applying your changes manually in HA and I now get: a 404 when it tries to load the places endpoint:
2023-12-20 18:42:32.127 DEBUG (MainThread) [life360.api] Error GET(https://api-cloudfront.life360.com/v4/circles/REDACTED/places), attempt 1: ClientResponseError(RequestInfo(url=URL('https://api-cloudfront.life360.com/v4/circles/REDACTED/places'), ... status=404, message='Not Found'

**Comment by dcmeglio** [comment]
The members and places endpoints might be in v3, not v4

**Comment by pantherale0** [comment]
They might be, I need to look over the proxy logs again to see.

I know it pulls the circles from v4, but some requests are v5, V6 etc (hence my comment in the core issue about the API being messy).

I'm hoping if we can get this to work with a recent version of the mobile app, we'll stand a good chance "long term".

**Comment by dcmeglio** [comment]
I added the following and it now loads properly in HA. Obviously I haven't tested every workflow but it initializes and detects location status again.

```
_CIRCLE_URL_FMT_V3 = f"{_BASE_CMD_V3}circles/{{circle_id}}"
_CIRCLE_MEMBERS_URL_FMT = f"{_CIRCLE_URL_FMT_V3}/members"
_CIRCLE_PLACES_URL_FMT = f"{_CIRCLE_URL_FMT_V3}/places"
```

**Comment by JeedHome44** [comment]
> I added the following and it now loads properly in HA. Obviously I haven't tested every workflow but it initializes and detects location status again.
> 
> 
> 
> ```
> 
> _CIRCLE_URL_FMT_V3 = f"{_BASE_CMD_V3}circles/{{circle_id}}"
> 
> _CIRCLE_MEMBERS_URL_FMT = f"{_CIRCLE_URL_FMT_V3}/members"
> 
> _CIRCLE_PLACES_URL_FMT = f"{_CIRCLE_URL_FMT_V3}/places"
> 
> ```

good job! It's work for me too !

**Comment by pnbruckner** [comment]
@dcmeglio thank you!

Starting from the last commit from @pantherale0 in this PR, and changing only this line back to V3 seems to make everything work:
```python
_CIRCLE_URL_FMT = f"{_BASE_CMD_V3}circles/{{circle_id}}"
```
I'll suggest that change in the PR review.

File: life360/api.py [hunk 1]
```diff
@@ -12,10 +12,11 @@
...
 _BASE_URL = f"{_PROTOCOL}{_HOST}"
-_BASE_CMD = f"{_BASE_URL}/v3/"
-_TOKEN_URL = f"{_BASE_CMD}oauth2/token.json"
-_CIRCLES_URL = f"{_BASE_CMD}circles.json"
-_CIRCLE_URL_FMT = f"{_BASE_CMD}circles/{{circle_id}}"
+_BASE_CMD_V3 = f"{_BASE_URL}/v3/"
+_BASE_CMD_V4 = f"{_BASE_URL}/v4/"
+_TOKEN_URL = f"{_BASE_CMD_V3}oauth2/token"
+_CIRCLES_URL = f"{_BASE_CMD_V4}circles"
+_CIRCLE_URL_FMT = f"{_BASE_CMD_V4}circles/{{circle_id}}"
```
Comment (pnbruckner) [inline life360/api.py:19]: Try changing this back to V3:
```python
_CIRCLE_URL_FMT = f"{_BASE_CMD_V3}circles/{{circle_id}}"
```

Comment (dcmeglio) [reply life360/api.py:19]: Yup just tested and that works for me

**Review by pantherale0 — COMMENTED** [review]
Resolved - we'll just get it working for now.

**Comment by dmtalon** [comment]
For those on a Rpi's, the api.py location is  /homeassistant/life360.  I swapped in the new api.py, restarted HA and was able to reconnect.

I'm assuming the other locations refer to the docker installs.

**Comment by jcarlier** [comment]
if I understood correctly, while waiting for 2023.12.4 you need to restore life360 (https://github.com/pantherale0/life360/tree/a8acbefdd3b75edd6af6e7df81f3a57649ef21af) in the /homeassistant directory (same level as 'custom_components') and then restart??????

**Comment by pdecker414** [comment]
> if I understood correctly, while waiting for 2023.12.4 you need to restore life360 (https://github.com/pantherale0/life360/tree/a8acbefdd3b75edd6af6e7df81f3a57649ef21af) in the /homeassistant directory (same level as 'custom_components') and then restart??????

I made the changes as listed and got mine working again until they release so I am happy. It really wasn't too many lines to fix. Thank you to the Dev for working on this so quickly. I use it to keep an eye on my teanage son with notifications if he goes outside of areas we have allowed.

**Comment by pnbruckner** [comment]
@jcarlier there are a couple of different workarounds that have been suggested. This is not the place to ask about those. Please check the HA community forum or discord. There are plenty of posts with details and people who can help with your exact install configuration.

**Pull request merged**